app/src/main/java/org/thoughtcrime/securesms/video/videoconverter/AudioTrackConverter.java

```diff
@@ -0,0 +1,421 @@
+        if (audioCodecInfo == null) {
+            // Don't fail CTS if they don't have an AAC codec (not here, anyway).
+            Log.e(TAG, "Unable to find an appropriate codec for " + OUTPUT_AUDIO_MIME_TYPE);
+            throw new FileNotFoundException();
+        }
+        if (VERBOSE) Log.d(TAG, "audio found codec: " + audioCodecInfo.getName());
+
+        final MediaFormat inputAudioFormat = mAudioExtractor.getTrackFormat(audioInputTrack);
+        mInputDuration = inputAudioFormat.containsKey(MediaFormat.KEY_DURATION) ? inputAudioFormat.getLong(MediaFormat.KEY_DURATION) : 0;
+
+        final MediaFormat outputAudioFormat =
+                MediaFormat.createAudioFormat(
+                        OUTPUT_AUDIO_MIME_TYPE,
+                        inputAudioFormat.getInteger(MediaFormat.KEY_SAMPLE_RATE),
+                        inputAudioFormat.getInteger(MediaFormat.KEY_CHANNEL_COUNT));
+        outputAudioFormat.setInteger(MediaFormat.KEY_BIT_RATE, audioBitrate);
+        outputAudioFormat.setInteger(MediaFormat.KEY_AAC_PROFILE, OUTPUT_AUDIO_AAC_PROFILE);
+        outputAudioFormat.setInteger(MediaFormat.KEY_MAX_INPUT_SIZE, 16 * 1024);
+
+        // Create a MediaCodec for the desired codec, then configure it as an encoder with
+        // our desired properties. Request a Surface to use for input.
+        mAudioEncoder = createAudioEncoder(audioCodecInfo, outputAudioFormat);
+        // Create a MediaCodec for the decoder, based on the extractor's format.
+        mAudioDecoder = createAudioDecoder(inputAudioFormat);
+
+        mAudioDecoderInputBuffers = mAudioDecoder.getInputBuffers();
+        mAudioDecoderOutputBuffers = mAudioDecoder.getOutputBuffers();
+        mAudioEncoderInputBuffers = mAudioEncoder.getInputBuffers();
+        mAudioEncoderOutputBuffers = mAudioEncoder.getOutputBuffers();
+        mAudioDecoderOutputBufferInfo = new MediaCodec.BufferInfo();
+        mAudioEncoderOutputBufferInfo = new MediaCodec.BufferInfo();
+
+        if (mTimeFrom > 0) {
+            mAudioExtractor.seekTo(mTimeFrom * 1000, MediaExtractor.SEEK_TO_PREVIOUS_SYNC);
+            Log.i(TAG, "Seek audio:" + mTimeFrom + " " + mAudioExtractor.getSampleTime());
+        }
+    }
+
+    void setMuxer(final @NonNull Muxer muxer) throws IOException {
+        mMuxer = muxer;
+        if (mEncoderOutputAudioFormat != null) {
+            Log.d(TAG, "muxer: adding audio track.");
+            if (!mEncoderOutputAudioFormat.containsKey(MediaFormat.KEY_BIT_RATE)) {
+                mEncoderOutputAudioFormat.setInteger(MediaFormat.KEY_BIT_RATE, mAudioBitrate);
+            }
+            if (!mEncoderOutputAudioFormat.containsKey(MediaFormat.KEY_AAC_PROFILE)) {
+                mEncoderOutputAudioFormat.setInteger(MediaFormat.KEY_AAC_PROFILE, OUTPUT_AUDIO_AAC_PROFILE);
+            }
+            mOutputAudioTrack = muxer.addTrack(mEncoderOutputAudioFormat);
+        }
+    }
+
+    void step() throws IOException {

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 55. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 150-330 into a separate method.